### PR TITLE
[Event Hubs] Eventhubs bug fix - sync in memory checkpoint is not in use

### PR DESCRIPTION
**Pull request by YijunXieMS**

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py [hunk 1]
```diff
@@ -71,8 +71,8 @@ def __init__(
             "on_partition_close", None
         )  # type: Optional[Callable[[PartitionContext, CloseReason], None]]
         self._checkpoint_store = kwargs.get(
-            "checkpoint_store", InMemoryCheckpointStore()
-        )  # type: Optional[CheckpointStore]
+            "checkpoint_store"
```
Comment (ellismg) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py]: For my own understanding - what's the purpose of this change.  It the goal to just ensure that if someone sets a falsey value for the kwarg we don't try to use it and instead create the `InMemoryCheckpointStore` object?  Is it something to do with the the fact that this allows us to delay construction of the `InMemoryCheckpointStore()` in cases where we didn't need to use it?  Something else?

Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py]: Hey @ellismg ,

Make long story short:
With the `InMemoryCheckpointStore`, users can call `update_checkpoint` now even if they don't provide a checkpoint_store, and as its name shows, it keeps in memory checkpoint which would be used for internal receiver recovery from error.

----- Long story:

Previously if a user doesn't provide a `checkpoint_store`, the sdk would simply log a warning  when `update_checkpoint` called. And if error is raised during the receiving process, it would recover receiving from the starting position.

After a discussion with architects and JS team, we've decided to provide the ability to do in memory checkpoint which can be used for internal partition receiver recovery and consistent user experience on the checkpoint stuff.

Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py]: Hey @ellismg 

----- long story short
With the InMemoryCheckpointStore, users can call update_checkpoint now even if they don't provide a checkpoint_store, and as its name shows, it keeps in memory checkpoint which would be used for internal receiver recovery from error.

----- Backgrounds:

Previously if a user doesn't provide a checkpoint_store, the sdk would simply log a warning when update_checkpoint called. And if error is raised during the receiving process, it would recover receiving from the starting position.

After a discussion with architects and JS team, we've decided to provide the ability to do in memory checkpoint which can be used for internal partition receiver recovery and consistent user experience on the checkpoint stuff.

Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py]: Thank Adam. Just to supplement one thing, this InMemoryCheckpointStore is an internal class, not exposed to users.

Comment (ellismg) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py]: Thanks for the context @yunhaoling!  The thing I was missing here was that `EventHubConsumerClient` internally creates an `EventProcessor`, and when doing so it explicitly sets the `checkpoint_store` keyword argument to `self._checkpoint_store` which itself was initialized by `self._checkpoint_store = kwargs.pop("checkpoint_store", None)`, so it is possible that the kwarg here could be set, but would be set to a value of `None`.  With the old code, since the value is present in the dictionary, `kawgs.get("checkpoint_store", InMemoryCheckpointStore())` would return None (since the value is set, it's just set to None) we not pick up the `InMemoryCheckpointStore()`.

Thanks for helping me understand!
